### PR TITLE
Added commands "Go Lint Package" and "Go Lint Workspace"

### LINKED ISSUE #1041: Calling lint and vet tool by hotkey or from interface
Great plugin, but calling lint and vet tools manualy sometimes needed to avoid freezings on save with autolint/autovet enabled.

### PR DESCRIPTION
**Pull request by ndrewnee**
Fixes #1041

### CODE REVIEW

**Comment by ndrewnee** [comment]
I don't know why happened this [job error](https://travis-ci.org/Microsoft/vscode-go/jobs/286677054), so reopening pull request to trigger travis

File: src/goMain.ts [hunk 1]
```diff
@@ -289,6 +290,30 @@ export function activate(ctx: vscode.ExtensionContext): void {
 
 	ctx.subscriptions.push(vscode.commands.registerCommand('go.get.package', goGetPackage));
 
+	ctx.subscriptions.push(vscode.commands.registerCommand('go.lint.package', () => {
+		let editor = vscode.window.activeTextEditor;
+		let documentUri = editor ? editor.document.uri : null;
+		let goConfig = vscode.workspace.getConfiguration('go', documentUri);
+
```
Comment (ramya-rao-a) [inline src/goMain.ts]: let's move all this logic inside `lintCurrentPackage` function. This is to be consistent with all the other commands and to keep the `goMain.ts` file clear
Same for `lintWorkspace` function

File: src/goMain.ts [hunk 2]
```diff
@@ -289,6 +290,30 @@ export function activate(ctx: vscode.ExtensionContext): void {
...
+		lintCurrentPackage(documentUri, goConfig)
+			.then(handleDiagnosticErrors(editor ? editor.document : null))
```
Comment (ramya-rao-a) [inline src/goMain.ts]: calling `handleDiagnosticErrors` clears any existing build and vet errors. It should only update the warnings.

**Comment by ndrewnee** [comment]
@ramya-rao-a No problem) I merged master and made some changes, you can view it. Thanks for your review)

**Comment by ramya-rao-a** [comment]
The diagnostics from build/lint/vet on save features were not showing up, fixed with https://github.com/Microsoft/vscode-go/pull/1258/commits/99f0957a247225f2bde70e3a70c11d88ff66b644

The linting result from the command was removing the build errors from before, fixed with https://github.com/Microsoft/vscode-go/pull/1258/commits/ef737c9891cfde5ed879a08aac50d5c3129f07af

**Comment by ramya-rao-a** [comment]
Merged, thanks @ndrewnee!

**Pull request merged**